### PR TITLE
Update bundle to v0.3.0 and cherry-pick commits from master

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
1. Updates bundle and manifests to v0.3.0
2.  Cherry-picks the following commits from master.
    #548 - Improve docs for custom cluster roles
    #514 - add route/ingress url to .status
    #553 - fix argocd-tls-cert-cm is overriden by any change
    #554 - add docs for hosts field in .status
    #557 - remove unnecessary roles/bindings for target namespace
    #560 - bump argocd dependency from 2.2.2 to 2.2.4
    #549 - update operator capability level 

**Have you updated the necessary documentation?**
* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:
Since this is a clean(no-merge conflicts) cherry-picking, no additional testing is required.

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
Authors of the cherry-picked commits are 
@reginapizza @chetan-rns @iam-veeramalla @ciiay @iam-veeramalla @jaideepr97 , PTAL.

File: bundle/manifests/argocd-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -124,15 +124,15 @@ metadata:
           }
         }
       ]
-    capabilities: Auto Pilot
+    capabilities: Deep Inisights
```
Comment (jannfis) [inline bundle/manifests/argocd-operator.clusterserviceversion.yaml:127]:
```suggestion
    capabilities: Deep Insights
```

File: config/manifests/bases/argocd-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ kind: ClusterServiceVersion
 metadata:
   annotations:
     alm-examples: '[]'
-    capabilities: Auto Pilot
+    capabilities: Deep Inisights
```
Comment (jannfis) [inline config/manifests/bases/argocd-operator.clusterserviceversion.yaml:6]:
```suggestion
    capabilities: Deep Insights
```

File: deploy/olm-catalog/argocd-operator/0.2.0/argocd-operator.v0.2.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -124,7 +124,7 @@ metadata:
           }
         }
       ]
-    capabilities: Auto Pilot
+    capabilities: Deep Inisights
```
Comment (jannfis) [inline deploy/olm-catalog/argocd-operator/0.2.0/argocd-operator.v0.2.0.clusterserviceversion.yaml:127]:
```suggestion
    capabilities: Deep Insights
```

File: config/manifests/bases/argocd-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -660,5 +660,5 @@ spec:
   maturity: alpha
   provider:
     name: Argo CD Community
-  replaces: argocd-operator.v0.1.0
-  version: 0.2.0
+  replaces: argocd-operator.v0.2.0
```
Comment (jannfis) [inline config/manifests/bases/argocd-operator.clusterserviceversion.yaml:663]: Going forward, we may consider using `skipRanges` in the Argo CD Operator's CSV as well?

**Review by jannfis — APPROVED** [review]
LGTM with a couple of nits.

**Pull request merged**